include/boost/graph/link_cut_trees.hpp

```diff
@@ -0,0 +1,283 @@
+//
+//=======================================================================
+// Copyright 2019
+// Author: Yi Ji
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//=======================================================================
+//
+#ifndef BOOST_LINK_CUT_TREES_HPP
+#define BOOST_LINK_CUT_TREES_HPP
+
+#include <boost/graph/properties.hpp>
+#include <boost/unordered_map.hpp>
+
+namespace boost
+{
+    template<class ElementParentMap, class ElementChildMap = ElementParentMap>
+    class link_cut_trees
+    {
+    public:
+        inline link_cut_trees(ElementParentMap p, ElementChildMap l, ElementChildMap r) : parent(p), left(l), right(r) {}
+        
+        inline link_cut_trees(const link_cut_trees &c)
+            : parent(c.parent), left(c.left), right(c.right) {}
+
+        template <class Element>
+        inline void make_tree(Element x)
+        {
+            make_path(x);
+        }
```

Comment:
Hey, I was just wondering... do you actually need to use the `inline` keyword on all these member functions? 
I'm not sure it has any effect, because I think they are all inline by virtue of being defined _in situ_.
It will just simplify reading the code a bit if we can remove them.

Reply:
Yeah I agree, I have them removed now, thanks.
The CI builds are failing, but the error logs seem totally unrelated. 
Even a blank change failed to build: https://github.com/boostorg/graph/pull/243

Reply:
Dang, I can't help with the CI stuff, sorry.

Reply:
Yeah, could be that the develop branch needs some cleaning up before anyone can merge new features.

Reply:
No problem, CI builds passed now